packages/airnode-validator/src/config/config.ts

```diff
@@ -153,39 +157,66 @@ export const chainOptionsSchema = z
   })
   .strict();
 
+const defaultAirnodeRrp: SuperRefinement<{
+  id?: ChainId;
+  chainId?: ChainId;
+  contracts: SchemaType<typeof airnodeRrpContractSchema>;
+}> = (chainConfig, ctx) => {
+  if (chainConfig.contracts.AirnodeRrp === '') {
+    // id OR chainId refers to the chain ID depending on the calling schema
+    const id = (chainConfig.id || chainConfig.chainId)!;
+    if (!AirnodeRrpV0[id]) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message:
+          `AirnodeRrp contract address must be specified for chain ID '${id}'` +
+          `as there was no deployment for this chain found in @airnode/protocol/deployments/references.json`,
+        path: ['contracts'],
+      });
+      return;
+    }
+    // Default to the deployed AirnodeRrp contract address if contracts is not specified
+    chainConfig.contracts = { AirnodeRrp: AirnodeRrpV0[id] };
```

Comment:
This seems to me is not the way refinement should be used (to mutate to object). Am I wrong about this?